### PR TITLE
Update s/SpokeCluster/ManagedCluster/g

### PR DESCRIPTION
**Pull request by mdelder**
+ Update cluster/v1 API to be consistent with ManagedCluster naming.
+ Minor edits to documentation

### CODE REVIEW

File: operator/v1/types.go [hunk 1]
```diff
@@ -93,37 +93,37 @@ type KlusterletSpec struct {
...
-	// If it is set empty, spoke cluster has no externally accessible url that hub cluster can visit.
+	// If it is set empty, managed cluster has no externally accessible url that hub cluster can visit.
 	// +optional
 	ExternalServerURLs []ServerURL `json:"externalServerURLs,omitempty" protobuf:"bytes,5,opt,name=externalServerURLs"`
 }
 
 // ServerURL represents the apiserver url and ca bundle that is accessible externally
 type ServerURL struct {
-	// URL is the url of apiserver endpoint of the spoke cluster.
+	// URL is the url of apiserver endpoint of the manageed cluster.
```
Comment (pmorie) [inline operator/v1/types.go]: typo: managed

File: work/v1/types.go [hunk 2]
```diff
@@ -125,40 +125,40 @@ type AppliedManifestResourceMeta struct {
 	Namespace string `json:"namespace" protobuf:"bytes,5,opt,name=namespace"`
 }
 
-// ManifestWorkStatus represents the current status of spoke manifest workload
+// ManifestWorkStatus represents the current status of managed manifest workload
```
Comment (pmorie) [inline work/v1/types.go]: Want to make sure this is intentional, reads a little awkwardly. `managed cluster`?

File: operator/v1/types.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ type Klusterlet struct {
 type KlusterletSpec struct {
 	// Namespace is the namespace to deploy the agent.
 	// The namespace must have a prefix of "open-cluster-management-", and if it is not set,
-	// the namespace of "open-cluster-management-spoke" is used to deploy agent.
+	// the namespace of "open-cluster-management-agent" is used to deploy agent.
```
Comment (pmorie) [inline operator/v1/types.go:84]: reminder to plumb this into the registration operator repo when you get there

File: operator/v1/types.go [hunk 4]
```diff
@@ -93,37 +93,37 @@ type KlusterletSpec struct {
 	// +required
 	WorkImagePullSpec string `json:"workImagePullSpec,omitempty" protobuf:"bytes,3,opt,name=workImagePullSpec"`
 
-	// ClusterName is the name of the spoke cluster to be created on hub.
-	// The spoke agent generates a random name if it is not set, or discovers the appropriate cluster name on openshift.
+	// ClusterName is the name of the managed cluster to be created on hub.
+	// The klusterlet agent generates a random name if it is not set, or discovers the appropriate cluster name on openshift.
```
Comment (pmorie) [inline operator/v1/types.go]: We should decide now whether we want to conventionally capitalize `klusterlet` or not.

**Review by pmorie — COMMENTED** [review]
Found a couple nits

File: operator/v1/types.go (hunk 4 above)
Comment (mdelder) [reply operator/v1/types.go]: Good call. I opted for `Klusterlet`.

File: cluster/v1/types.go [hunk 5]
```diff
@@ -9,107 +9,108 @@ import (
...
 }
 
-// SpokeClusterSpec provides the information to securely connect to a remote server
+// ManagedClusterSpec provides the information to securely connect to a remote server
 // and verify its identity.
-type SpokeClusterSpec struct {
-	// SpokeClientConfigs represents a list of the apiserver address of the spoke cluster.
-	// If it is empty, spoke cluster has no accessible address to be visited from hub.
+type ManagedClusterSpec struct {
+	// ManagedClientConfigs represents a list of the apiserver address of the managed cluster.
```
Comment (qiujian16) [inline cluster/v1/types.go]: ManagedClusterClientConfig?

**Comment by itdove** [comment]
@mdelder  FYI This changes will request cycles in the rcm-controller because:
1) The CRD changes and the type.go change... so need to adapt the code.
2) In the code to be coherent we use the word "spoke" at many places (from Go variables to Go files) and so all need to be adapted to have a clear and compressible code.

**Comment by mdelder** [comment]
@itdove Agreed. There is a PR in flight to adopt the API updates in the `nucleus` operator (where the repo will also be renamed).

**Comment by pmorie** [comment]
> In the code to be coherent we use the word "spoke" at many places (from Go variables to Go files) and so all need to be adapted to have a clear and compressible code.

@itdove IMO there is no need to change comments, I think the meaning will remain clear.

**Comment by itdove** [comment]
I guess also issues must be open for each squad to make this change and also for the doc team. For me it is not very nice to have different names for the same thing, it just drive to confusion. Also because it is not impossible that in the future that name will change again ;-)

**Comment by mdelder** [comment]
@qiujian16 Updated `ManagedClientConfig` to `ManagedClusterClientConfig`. Also caught there the CRD filename had been missed.

I believe this PR is ready to be merged, barring any additional feedback from reviewers.

**Comment by qiujian16** [comment]
/assign pmorie
/assign deads2k

/lgtm 
leave to @pmorie @deads2k for any more comments.

**Comment by pmorie** [comment]
> I guess also issues must be open for each squad to make this change and also for the doc team. For me it is not very nice to have different names for the same thing, it just drive to confusion. 

I totally get it. I think that 'spoke' is a meaningful var / param name to use, whatever the name of the API resource, and I wouldn't feel like it was wrong to see those terms in code. I'll probably continue to say 'spoke' in conversation because it is specific and shorter than the API resource name.

> Also because it is not impossible that in the future that name will change again ;-)

Totally possible, all the more reason to avoid exhaustively rooting it out in places like code where it's fine as-is.

**Comment by mdelder** [comment]
@pmorie Are you okay to approve this PR?

**Pull request merged**